src/rgw/rgw_iam_policy.h

```diff
@@ -359,6 +360,18 @@ struct Condition {
     }
   };
 
+  struct ci_string_like : public std::binary_function<const std::string,
+                                                     const std::string,
+                                                     bool> {
+    bool operator ()(const std::string& s1,
+                    const std::string& s2) const {
+      if(fnmatch(s2.c_str(), s1.c_str(), FNM_CASEFOLD) == 0){
+        return true;
```

Comment:
hi @adamemerson , i test on amazon s3. 
the test policy set as follow
```
{
  "Id": "Policy1497346096739",
  "Version": "2012-10-17",
  "Statement": [
    {
      "Sid": "Stmt1497346083540",
      "Action": [
        "s3:GetObject"
      ],
      "Effect": "Allow",
      "Resource": "arn:aws:s3:::yly-policy/*",
      "Condition": {
        "StringLike": {
          "aws:Referer": "http://WWW.example.com/*"
        }
      },
      "Principal": "*"
    }
  ]
}
```
and curl with referer `http://www.example.com/`

```
> GET /yly-policy/1.txt HTTP/1.1
> User-Agent: curl/7.29.0
> Host: s3.amazonaws.com
> Accept: */*
> referer: http://www.example.com/
> 
< HTTP/1.1 403 Forbidden
< x-amz-request-id: FC9CE4656FA773F7
< x-amz-id-2: 2j6il8iRZowlAvP4ylLP2XFGS2ZAIpdZ9Mfb6JnQxiTts1SHnw6fEU6bk+jDmd0Rq6yK/3zBR98=
< Content-Type: application/xml
< Transfer-Encoding: chunked
< Date: Sat, 17 Jun 2017 13:02:09 GMT
< Server: AmazonS3

```


and curl with refer `http://WWW.example.com/`
```
> GET /yly-policy/1.txt HTTP/1.1
> User-Agent: curl/7.29.0
> Host: s3.amazonaws.com
> Accept: */*
> referer: http://WWW.example.com/
> 
< HTTP/1.1 200 OK
< x-amz-id-2: iuOA7wDe/OQoukdM+2MepD9g99GuP0Vkm2raDIYNSZSTt6M3AnDI+ztNX/5yDrr1+Tjt0cT1t/M=
< x-amz-request-id: AF2E4CA0DA7C49B2
< Date: Sat, 17 Jun 2017 13:02:48 GMT
< Last-Modified: Tue, 13 Jun 2017 08:43:37 GMT
< ETag: "70e2c79c9f37189d91b784edb36077ce"
< Accept-Ranges: bytes
< Content-Type: binary/octet-stream
< Content-Length: 273
< Server: AmazonS3

```

so i wonder why we need  get case sensitivity ?